sql/core/src/main/scala/org/apache/spark/sql/execution/command/functions.scala

```diff
@@ -46,26 +46,47 @@ case class CreateFunctionCommand(
     functionName: String,
     className: String,
     resources: Seq[FunctionResource],
-    isTemp: Boolean)
+    isTemp: Boolean,
+    ifNotExists: Boolean,
+    replace: Boolean)
   extends RunnableCommand {
 
+  if (ifNotExists && replace) {
+    throw new AnalysisException("CREATE FUNCTION with both IF NOT EXISTS and REPLACE" +
+      " is not allowed.")
+  }
+
+  // Disallows 'CREATE TEMPORARY FUNCTION IF NOT EXISTS' to be consistent
+  // with 'CREATE TEMPORARY FUNCTION'
```

Comment:
Ah, do you mean that the value of the `overrideIfExists `parameter can be `ifnotexists `or `replace` ?